### PR TITLE
Fix related products E2Es for WC9.5

### PR DESCRIPTION
**Pull request by tomalec**
### Changes proposed in this Pull Request:

This PR adjusts E2E tests for WC 9.5 changes.

In WC 9.5, by default, there is no related products section on a single product page - I made the test skip in such a case.
Also, I added a "related" tag to all simple products to make sure simple products will appear in the related products section instead of "Cat teasers" - without the "Add to cart" button.

### Screenshots:

image

### Detailed test instructions:

1. Run E2E tests locally
2. Run [E2E workflow](https://github.com/woocommerce/google-listings-and-ads/actions/workflows/e2e-tests.yml) with `9.5.0-rc.1` - https://github.com/woocommerce/google-listings-and-ads/actions/runs/12361402622
3. Run [E2E workflow](https://github.com/woocommerce/google-listings-and-ads/actions/workflows/e2e-tests.yml) with `9.4.0` - https://github.com/woocommerce/google-listings-and-ads/actions/runs/12361396232
4. Review the code to check the tests are testing what they should

### Changelog entry

> Dev - Improve E2E tests for related products.

### CODE REVIEW

**Comment by tomalec** [comment]
See also https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/481

File: tests/e2e/utils/block-page.js [hunk 1]
```diff
@@ -41,14 +42,32 @@ export async function createPage( title, content ) {
 
 /**
  * Creates a shop page using blocks.
+ * @return {string} Slug of the created page.
```
Comment (eason9487) [inline tests/e2e/utils/block-page.js]: The type seems to be `Promise<string>`.

File: tests/e2e/utils/block-page.js [hunk 2]
```diff
@@ -41,14 +42,32 @@ export async function createPage( title, content ) {
...
 		await createPage( title, content );
 	}
+	return slug;
+}
+
+/**
+ * Creates a page with related products block.
+ *
+ * @param {number} productId Product ID.
+ * @return {string} Slug of the created page.
```
Comment (eason9487) [inline tests/e2e/utils/block-page.js]: The type seems to be `Promise<string>`.

**Review by eason9487 — APPROVED** [review]
Thanks for the compatibility fix! LGTM.

**Pull request merged**